### PR TITLE
Cherry-pick OpenSSL 3.4 assembly changes

### PR DESCRIPTION
**Pull request by botovq**
I need these commits to run the quictls test suite on a modern OpenBSD/amd64 laptop.

## Checklist
Thank you for your contribution. Please answer the following:

- [x] I acknowledge that I am authorized to submit this code under
the terms of the [Apache License](https://www.apache.org/licenses/LICENSE-2.0)
- [ ] All new public APIs are documented
- [ ] All new public functions have tests
- [ ] All new public functions have fuzzing tests

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -50,3 +50,7 @@ made private.
 
 - Simplified `util/shlib_wrap`; if your shell doesn't do "$@" properly
 when there are no arguments, please look at an earlier version.
+
+- Cherry-picked changes to assembly generation to support running code
```
Comment (richsalz) [inline CHANGES.md:54]: is it worth saying the original source of the cherry-pick?

File: crypto/perlasm/x86_64-xlate.pl [hunk 2]
```diff
@@ -156,6 +156,65 @@
...
+#	; follows:
+#	;	('.text', '.text')
+# .rodata
+#	; pop() so we can generate proper 'ENDS' for masm.
+#	; stack looks like:
+#	; 	('.text')
+#	; push '.rodata', so we can create corresponding ENDS for masm.
+#	; stack looks like:
+#	;	('.rodata', '.text')
+# .previous
```
Comment (richsalz) [inline crypto/perlasm/x86_64-xlate.pl:206]: that's a cool construct.

**Review by richsalz — APPROVED** [review]
one comment, okay to merge if you want to fill it in later (or not at all)

File: CHANGES.md (hunk 1 above)
Comment (botovq) [reply CHANGES.md:54]: Yes, it's probably better to elaborate on the provenance. I'll rephrase this when I pull the corresponding arm64 changes.

**Pull request merged**